x-pack/plugins/security/public/authentication/overwritten_session/overwritten_session_page.tsx

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useEffect, useState } from 'react';
+import ReactDOM from 'react-dom';
+import { EuiButton } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { CoreStart, IBasePath } from 'src/core/public';
+import { AuthenticationServiceSetup } from '../authentication_service';
+import { AuthenticationStatePage } from '../components';
+
+interface Props {
+  basePath: IBasePath;
+  authc: AuthenticationServiceSetup;
+}
+
+export function OverwrittenSessionPage({ authc, basePath }: Props) {
+  const [username, setUsername] = useState<string | null>(null);
+  useEffect(() => {
+    authc.getCurrentUser().then(user => setUsername(user.username));
+  }, [authc]);
+
+  if (username == null) {
+    return null;
+  }
```

Comment:
This isn't currently covered by any unit tests.

What do you think about abstracting the username out to the OverwrittenSessionApp and passing it into the OverwrittenSessionPage as a property?

Reply:
> This isn't currently covered by any unit tests.

Will add!

> What do you think about abstracting the username out to the OverwrittenSessionApp and passing it into the OverwrittenSessionPage as a property?

Honestly I'm torn on this, I don't really want to move this logic to `overwrittenSessionApp` that is supposed to be as dumb as possible or making it's a part of `renderOverwrittenSessionPage` that's harder to test (because it hides `ReactDOM.render`). Do you have some benefit in mind that I'm missing?

Reply:
> Honestly I'm torn on this, I don't really want to move this logic to `overwrittenSessionApp` that is supposed to be as dumb as possible or making it's a part of `renderOverwrittenSessionPage` that's harder to test (because it hides `ReactDOM.render`). Do you have some benefit in mind that I'm missing?
It made sense in my head while I was reviewing it, but I don't think there's any benefit tbh. Please disregard!